### ISSUE TITLE
401 Unauthorized on RUM POST

### ISSUE THREAD

**Issue by kanadaj** [elastic/apm-server#6757]
**APM Server version** (`apm-server version`):
apm-server version 7.15.2 (amd64), libbeat 7.15.1 [0702551667e5319822d00dee613ea2a1501615bf built 2021-11-04 13:10:15 +0000 UTC]

**Description of the problem including expected versus actual behavior**: 
When using `@elastic/apm-rum` 5.9.1, requests to `https://domain.com/intake/v2/rum/events` and `https://domain.com/intake/v3/rum/events`  seem to fail with a HTTP 401:
```json
{
  "error": "authentication failed: missing or improperly formatted Authorization header: expected 'Authorization: Bearer secret_token' or 'Authorization: ApiKey base64(API key ID:API key)'"
}
```

**Steps to reproduce**:

 1. Install Elastic stack 7.15.2 with APM server
 2. Configure APM Server for RUM:
 ```yaml
    apm-server:
      auth:
        anonymous.allow_service:
          - Frontend
        anonymous.rate_limit.event_limit: 300
        anonymous.rate_limit.ip_limit: 1000
      rum:
        allow_origins:
          - 'https://localhost:5001'
        enabled: true
        exclude_from_grouping: ^/webpack
        library_pattern: node_modules|bower_components|~
        source_mapping.cache.expiration: 5m
        source_mapping.enabled: true
        source_mapping.index_pattern: apm-*-sourcemap*
```
(Note: I've used ECK for this, the above is copied from my `ApmServer.spec.config`)
 3. Add `@elastic/apm-rum` to a front-end project and initialize it:
 ```ts
import { init as initApm } from '@elastic/apm-rum'

const apm = initApm({

    // Set required service name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
    serviceName: 'Frontend',
    
    // Set custom APM Server URL (default: http://localhost:8200)
    serverUrl: 'https://domain.com:443',

    // Set service version (required for sourcemap feature)
    serviceVersion: window.version,
})
```
4. Load the page.

**Provide logs (if relevant)**:

```
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.548Z","log.origin":{"file.name":"instance/beat.go","file.line":665},"message":"Home path: [/usr/share/apm-server] Config path: [/usr/share/apm-server] Data path: [/usr/share/apm-server/data] Logs path: [/usr/share/apm-server/logs]","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.556Z","log.origin":{"file.name":"instance/beat.go","file.line":673},"message":"Beat ID: 221ff5eb-8b89-472f-a79d-f8390efa9acd","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.557Z","log.logger":"seccomp","log.origin":{"file.name":"seccomp/seccomp.go","file.line":124},"message":"Syscall filter successfully installed","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.557Z","log.logger":"beat","log.origin":{"file.name":"instance/beat.go","file.line":1014},"message":"Beat info","service.name":"apm-server","system_info":{"beat":{"path":{"config":"/usr/share/apm-server","data":"/usr/share/apm-server/data","home":"/usr/share/apm-server","logs":"/usr/share/apm-server/logs"},"type":"apm-server","uuid":"221ff5eb-8b89-472f-a79d-f8390efa9acd"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.557Z","log.logger":"beat","log.origin":{"file.name":"instance/beat.go","file.line":1023},"message":"Build info","service.name":"apm-server","system_info":{"build":{"commit":"0702551667e5319822d00dee613ea2a1501615bf","libbeat":"7.15.1","time":"2021-11-04T13:10:15.000Z","version":"7.15.2"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.557Z","log.logger":"beat","log.origin":{"file.name":"instance/beat.go","file.line":1026},"message":"Go runtime info","service.name":"apm-server","system_info":{"go":{"os":"linux","arch":"amd64","max_procs":16,"version":"go1.16.6"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.558Z","log.logger":"beat","log.origin":{"file.name":"instance/beat.go","file.line":1030},"message":"Host info","service.name":"apm-server","system_info":{"host":{"architecture":"x86_64","boot_time":"2021-08-07T13:34:24Z","containerized":true,"name":"apmserver-apm-server-7f87cbd5bb-827bm","ip":["127.0.0.1/8","10.129.40.172/32"],"kernel_version":"5.4.0-70-generic","mac":["06:53:bb:f7:07:21"],"os":{"type":"linux","family":"redhat","platform":"centos","name":"CentOS Linux","version":"7 (Core)","major":7,"minor":9,"patch":2009,"codename":"Core"},"timezone":"UTC","timezone_offset_sec":0,"id":"a8bdf8c4d130d985eace1ff62bf02127"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.559Z","log.logger":"beat","log.origin":{"file.name":"instance/beat.go","file.line":1059},"message":"Process info","service.name":"apm-server","system_info":{"process":{"capabilities":{"inheritable":["chown","dac_override","fowner","fsetid","kill","setgid","setuid","setpcap","net_bind_service","net_raw","sys_chroot","mknod","audit_write","setfcap"],"permitted":null,"effective":null,"bounding":["chown","dac_override","fowner","fsetid","kill","setgid","setuid","setpcap","net_bind_service","net_raw","sys_chroot","mknod","audit_write","setfcap"],"ambient":null},"cwd":"/usr/share/apm-server","exe":"/usr/share/apm-server/apm-server","name":"apm-server","pid":1,"ppid":0,"seccomp":{"mode":"filter","no_new_privs":true},"start_time":"2021-11-30T21:26:25.690Z"},"ecs.version":"1.6.0"}}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.560Z","log.origin":{"file.name":"instance/beat.go","file.line":309},"message":"Setup Beat: apm-server; Version: 7.15.2","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"warn","@timestamp":"2021-11-30T21:26:26.561Z","log.logger":"cfgwarn","log.origin":{"file.name":"tlscommon/config.go","file.line":100},"message":"DEPRECATED: Treating the CommonName field on X.509 certificates as a host name when no Subject Alternative Names are present is going to be removed. Please update your certificates if needed. Will be removed in version: 8.0.0","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.561Z","log.logger":"esclientleg","log.origin":{"file.name":"eslegclient/connection.go","file.line":100},"message":"elasticsearch url: https://elasticsearch-es-http.elasticsearch.svc:9200","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.562Z","log.logger":"publisher","log.origin":{"file.name":"pipeline/module.go","file.line":113},"message":"Beat name: apmserver-apm-server-7f87cbd5bb-827bm","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.563Z","log.logger":"config","log.origin":{"file.name":"config/rum.go","file.line":89},"message":"Falling back to elasticsearch output for sourcemap storage","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.563Z","log.logger":"beater","log.origin":{"file.name":"beater/beater.go","file.line":562},"message":"Registering pipeline callback","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.563Z","log.origin":{"file.name":"instance/beat.go","file.line":473},"message":"apm-server start running.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.564Z","log.logger":"config","log.origin":{"file.name":"config/rum.go","file.line":89},"message":"Falling back to elasticsearch output for sourcemap storage","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.564Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":151},"message":"Listening on: [::]:8200","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.564Z","log.origin":{"file.name":"kibana/client.go","file.line":167},"message":"Kibana url: https://kibana-kb-http.elasticsearch.svc:5601","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.565Z","log.logger":"beater","log.origin":{"file.name":"apm-server/main.go","file.line":56},"message":"creating transaction metrics aggregation with config: {Enabled:true Interval:1m0s MaxTransactionGroups:10000 HDRHistogramSignificantFigures:2}","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.568Z","log.logger":"beater","log.origin":{"file.name":"apm-server/main.go","file.line":72},"message":"creating service destinations aggregation with config: {Enabled:true Interval:1m0s MaxGroups:10000}","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.568Z","log.logger":"agentcfg","log.origin":{"file.name":"agentcfg/cache.go","file.line":38},"message":"Cache creation with expiration 30s.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.origin":{"file.name":"kibana/client.go","file.line":167},"message":"Kibana url: https://kibana-kb-http.elasticsearch.svc:5601","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":123},"message":"Path / added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":123},"message":"Path /assets/v1/sourcemaps added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":123},"message":"Path /config/v1/agents added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":123},"message":"Path /config/v1/rum/agents added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":123},"message":"Path /intake/v2/rum/events added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":123},"message":"Path /intake/v3/rum/events added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":123},"message":"Path /intake/v2/events added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"handler","log.origin":{"file.name":"api/mux.go","file.line":123},"message":"Path /intake/v2/profile added to request handler","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"beater","log.origin":{"file.name":"beater/server.go","file.line":231},"message":"Starting apm-server [0702551667e5319822d00dee613ea2a1501615bf built 2021-11-04 13:10:15 +0000 UTC]. Hit CTRL-C to stop it.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":87},"message":"RUM endpoints enabled!","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"onboarding","log.origin":{"file.name":"beater/onboarding.go","file.line":34},"message":"Publishing onboarding document","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.569Z","log.logger":"beater","log.origin":{"file.name":"beater/http.go","file.line":107},"message":"SSL enabled.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.598Z","log.logger":"kibana","log.origin":{"file.name":"kibana/connecting_client.go","file.line":83},"message":"Successfully obtained connection to Kibana.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:26.611Z","log.logger":"kibana","log.origin":{"file.name":"kibana/connecting_client.go","file.line":83},"message":"Successfully obtained connection to Kibana.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.570Z","log.logger":"publisher_pipeline_output","log.origin":{"file.name":"pipeline/output.go","file.line":143},"message":"Connecting to backoff(elasticsearch(https://elasticsearch-es-http.elasticsearch.svc:9200))","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.570Z","log.logger":"publisher","log.origin":{"file.name":"pipeline/retry.go","file.line":219},"message":"retryer: send unwait signal to consumer","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.570Z","log.logger":"publisher","log.origin":{"file.name":"pipeline/retry.go","file.line":223},"message":"  done","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.579Z","log.logger":"esclientleg","log.origin":{"file.name":"eslegclient/connection.go","file.line":273},"message":"Attempting to connect to Elasticsearch version 7.15.1","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.596Z","log.logger":"esclientleg","log.origin":{"file.name":"eslegclient/connection.go","file.line":273},"message":"Attempting to connect to Elasticsearch version 7.15.1","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.610Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":86},"message":"Overwrite ILM setup is disabled. ","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.610Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":193},"message":"Set setup.template.name to 'apm-%{[observer.version]}'.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.610Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":195},"message":"Set setup.template.pattern to 'apm-%{[observer.version]}*'.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.670Z","log.origin":{"file.name":"template/load.go","file.line":111},"message":"Template \"apm-7.15.2\" already exists and will not be overwritten.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.670Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":205},"message":"Finished loading index template.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.673Z","log.logger":"index-management.ilm","log.origin":{"file.name":"ilm/std.go","file.line":170},"message":"ILM policy apm-rollover-30-days exists already.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.673Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":241},"message":"ILM policy apm-rollover-30-days successfully loaded.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.731Z","log.origin":{"file.name":"template/load.go","file.line":111},"message":"Template \"apm-7.15.2-profile\" already exists and will not be overwritten.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.731Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":222},"message":"Finished template setup for apm-7.15.2-profile.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.733Z","log.logger":"index-management.ilm","log.origin":{"file.name":"ilm/std.go","file.line":126},"message":"Index Alias apm-7.15.2-profile exists already.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.733Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":258},"message":"Write alias apm-7.15.2-profile successfully generated.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.796Z","log.origin":{"file.name":"template/load.go","file.line":111},"message":"Template \"apm-7.15.2-span\" already exists and will not be overwritten.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.796Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":222},"message":"Finished template setup for apm-7.15.2-span.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.799Z","log.logger":"index-management.ilm","log.origin":{"file.name":"ilm/std.go","file.line":126},"message":"Index Alias apm-7.15.2-span exists already.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.799Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":258},"message":"Write alias apm-7.15.2-span successfully generated.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.863Z","log.origin":{"file.name":"template/load.go","file.line":111},"message":"Template \"apm-7.15.2-transaction\" already exists and will not be overwritten.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.863Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":222},"message":"Finished template setup for apm-7.15.2-transaction.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.865Z","log.logger":"index-management.ilm","log.origin":{"file.name":"ilm/std.go","file.line":126},"message":"Index Alias apm-7.15.2-transaction exists already.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.865Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":258},"message":"Write alias apm-7.15.2-transaction successfully generated.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.926Z","log.origin":{"file.name":"template/load.go","file.line":111},"message":"Template \"apm-7.15.2-error\" already exists and will not be overwritten.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.926Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":222},"message":"Finished template setup for apm-7.15.2-error.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.929Z","log.logger":"index-management.ilm","log.origin":{"file.name":"ilm/std.go","file.line":126},"message":"Index Alias apm-7.15.2-error exists already.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.929Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":258},"message":"Write alias apm-7.15.2-error successfully generated.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.986Z","log.origin":{"file.name":"template/load.go","file.line":111},"message":"Template \"apm-7.15.2-metric\" already exists and will not be overwritten.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.986Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":222},"message":"Finished template setup for apm-7.15.2-metric.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.989Z","log.logger":"index-management.ilm","log.origin":{"file.name":"ilm/std.go","file.line":126},"message":"Index Alias apm-7.15.2-metric exists already.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.989Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":258},"message":"Write alias apm-7.15.2-metric successfully generated.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.989Z","log.logger":"index-management","log.origin":{"file.name":"idxmgmt/manager.go","file.line":123},"message":"Finished index management setup.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.990Z","log.logger":"pipelines","log.origin":{"file.name":"pipeline/register.go","file.line":55},"message":"Pipeline already registered: apm","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.991Z","log.logger":"pipelines","log.origin":{"file.name":"pipeline/register.go","file.line":55},"message":"Pipeline already registered: apm_data_stream_migration","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.992Z","log.logger":"pipelines","log.origin":{"file.name":"pipeline/register.go","file.line":55},"message":"Pipeline already registered: apm_user_agent","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.993Z","log.logger":"pipelines","log.origin":{"file.name":"pipeline/register.go","file.line":55},"message":"Pipeline already registered: apm_user_geo","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.994Z","log.logger":"pipelines","log.origin":{"file.name":"pipeline/register.go","file.line":55},"message":"Pipeline already registered: apm_ingest_timestamp","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.995Z","log.logger":"pipelines","log.origin":{"file.name":"pipeline/register.go","file.line":55},"message":"Pipeline already registered: apm_remove_span_metadata","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.996Z","log.logger":"pipelines","log.origin":{"file.name":"pipeline/register.go","file.line":55},"message":"Pipeline already registered: apm_error_grouping_name","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.997Z","log.logger":"pipelines","log.origin":{"file.name":"pipeline/register.go","file.line":55},"message":"Pipeline already registered: apm_metrics_dynamic_template","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.997Z","log.logger":"pipelines","log.origin":{"file.name":"pipeline/register.go","file.line":58},"message":"Registered Ingest Pipelines successfully.","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2021-11-30T21:26:27.997Z","log.logger":"publisher_pipeline_output","log.origin":{"file.name":"pipeline/output.go","file.line":151},"message":"Connection to backoff(elasticsearch(https://elasticsearch-es-http.elasticsearch.svc:9200)) established","service.name":"apm-server","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2021-11-30T21:27:55.406Z","log.logger":"request","log.origin":{"file.name":"middleware/log_middleware.go","file.line":60},"message":"authentication failed: missing or improperly formatted Authorization header: expected 'Authorization: Bearer secret_token' or 'Authorization: ApiKey base64(API key ID:API key)'","service.name":"apm-server","url.original":"/intake/v3/rum/events","http.request.method":"POST","user_agent.original":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.55 Safari/537.36 Edg/96.0.1054.34","source.address":"[REDACTED]","http.request.body.bytes":2465,"http.request.id":"1812e854-be04-45c8-aeba-157838a6b121","event.duration":140216,"http.response.status_code":401,"error.message":"authentication failed: missing or improperly formatted Authorization header: expected 'Authorization: Bearer secret_token' or 'Authorization: ApiKey base64(API key ID:API key)'","ecs.version":"1.6.0"}
```

**Comment by kanadaj** [comment]
Setting `apm-server.auth.anonymous.enabled` to true explicitly seems to fix this. It would seem that contrary to the documentation, this property isn't implicitly set by the agent.

**Comment by axw** [comment]
@kanadaj indeed, it is contrary to the docs - sorry for that. This has been fixed in https://github.com/elastic/apm-server/pull/6607. We're expecting 7.16.0 soon, and I think we may have missed the boat for getting this fix in, but we'll make sure it's in 7.16.1.

**Comment by simitt** [comment]
Fixes have been released with `7.16.1`.

**Comment by ppumkin** [comment]
Can confirm it has been fixed in 7.16.1
I was getting this error from FrontEnd. Went into ElasticCloud, upgraded to 7.16.1 as suggested here. 
Started accepting data. 
Thank you!
